lib/rust/ensogl/core/src/display/object/instance.rs

```diff
@@ -1,10 +1,248 @@
 //! Implementation of display objects, elements that have visual representation and can form
 //! hierarchical layouts. The implementation is very careful about performance, it tracks the
 //! transformation changes and updates only the needed subset of the display object tree on demand.
+//!
+//!
+//! # Auto Layout
+//!
+//! Display objects have built-in two-dimensional grid-based layout system, also called the Grid. It
+//! aims at providing an efficient way to lay out, align and distribute space among items in a
+//! container, even when their size is unknown and/or dynamic.
+//!
+//! The main idea behind the Grid layout system is to give the container the ability to alter its
+//! items’ width/height (and order) to best fill the available space (mostly to accommodate to all
+//! kind of display devices and screen sizes). A flex container expands items to fill available free
+//! space or shrinks them to prevent overflow.
+//!
+//! Most importantly, the layout system is direction-agnostic as opposed to common HTML layouts
+//! (block which is vertically-based and inline which is horizontally-based). While those work well
+//! for pages, they lack flexibility to support large or complex applications (especially when it
+//! comes to orientation changing, resizing, stretching, shrinking, etc.). The layout system
+//! described in this chapter can remind of CSS Flexbox and CSS Grid. In fact, it is heavily
+//! inspired by these systems, but it was extended in order to provide functionality of both of them
+//! in a single layout algorithm.
+//!
+//!
+//! ## Terminology and Concepts
+//! Before diving into the concepts of Grid it’s important to understand the terminology. Since the
+//! terms involved here are all kinda conceptually similar, it’s easy to confuse them with one
+//! another if you don’t first memorize their meanings defined by the Grid specification.
+//!
+//! - **Grid container.** The display object with enabled auto-layout. It is the element that
+//!   contains all the grid items (children of the display object).
+//!
+//! - **Grid item.** The child (i.e. direct descendants) of the grid container. A grid item can be
```

Comment:
Maybe it's just me, but I think `:` is more suitable here:
```suggestion
//! - **Grid item:** The child (i.e. direct descendants) of the grid container. A grid item can be
```